pkg/cloud/services/networking/floatingip.go

```diff
@@ -53,6 +56,15 @@ func (s *Service) GetOrCreateFloatingIP(openStackCluster *infrav1.OpenStackClust
 		return nil, err
 	}
 
+	if len(openStackCluster.Spec.Tags) > 0 {
+		_, err = attributestags.ReplaceAll(s.client, "floatingips", fp.ID, attributestags.ReplaceAllOpts{
```

Comment:
Could you add an observation call for the metric. Similar to https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/347f7c464c8bd8f131d02e77092afcd0988f32fb/pkg/cloud/services/compute/instance.go#L546-L550